hazelcast/src/test/java/com/hazelcast/cache/CacheCreationTest.java

```diff
@@ -0,0 +1,120 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache;
+
+import com.hazelcast.cache.impl.HazelcastServerCachingProvider;
+import com.hazelcast.config.Config;
+import com.hazelcast.config.XmlConfigBuilder;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.SlowTest;
+import org.junit.*;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import javax.cache.Cache;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class CacheCreationTest extends HazelcastTestSupport {
+
+    static Config hzConfig;
+    private static final int THREAD_COUNT = 4;
+
+    @BeforeClass
+    public static void init() throws Exception{
+        final URL configUrl1 = CacheCreationTest.class.getClassLoader().getResource("test-hazelcast-real-jcache.xml");
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(configUrl1.getFile());
+        hzConfig = configBuilder.build();
+    }
+
+    @Before
+    @After
+    public void killAllHazelcastInstances() throws Exception {
+        Hazelcast.shutdownAll();
+    }
+
+    @Test
+    public void creatingASingleCache() throws URISyntaxException {
+        HazelcastServerCachingProvider cachingProvider = createCachingProvider(hzConfig);
+        Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache" + 1);
+        cache.get(1);
+    }
+
+    @Test
+    public void creatingASingleCacheFromMultiProviders() throws URISyntaxException, InterruptedException {
+
+        ExecutorService execSer = Executors.newFixedThreadPool(THREAD_COUNT);
+
+        final CountDownLatch latch = new CountDownLatch(THREAD_COUNT);
+        for (int i = 0; i < THREAD_COUNT; i++) {
+            execSer.execute(new Runnable() {
+                @Override
+                public void run() {
+                    HazelcastServerCachingProvider cachingProvider = createCachingProvider(hzConfig);
+                    Cache<Object, Object> cache = cachingProvider.getCacheManager().getCache("xmlCache");
+                    cache.get(1);
+                    latch.countDown();
+                }
+            });
+        }
+        HazelcastTestSupport.assertOpenEventually(latch);
+        if (!execSer.awaitTermination(3, TimeUnit.SECONDS)) {
+            execSer.shutdown();
+        }
+    }
+
+    @Test
+    public void creatingMultiCacheFromMultiProviders() throws URISyntaxException, InterruptedException {
+
```

Comment:
In case of test I just say 'throws Exception' I don't care which exceptions are thrown. They are all going to cause the test to fail.